background.ts

```diff
@@ -576,46 +577,64 @@ async function getVersion() {
 }
 
 /**
- * assume sync activities aren't going to be costly for a UI app.
+ * Manages the state of integration symlinks.
  * @param name -- basename of the resource to link
- * @param state -- true to symlink, false to delete
+ * @param desiredPresent -- true to symlink, false to delete
  */
-async function linkResource(name: string, state: boolean): Promise<Error | null> {
+async function linkResource(name: string, desiredPresent: boolean): Promise<void> {
   const linkPath = path.join(paths.integration, name);
 
-  let err: Error | null = await new Promise((resolve) => {
-    fs.mkdir(paths.integration, { recursive: true }, resolve);
-  });
-
-  if (err) {
-    console.error(`Error creating the directory ${ paths.integration }: ${ err.message }`);
+  try {
+    await fsPromises.mkdir(paths.integration, { recursive: true });
+  } catch (error: any) {
+    console.error(`Error creating integrations directory ${ paths.integration }: ${ error.message }`);
+  }
 
-    return err;
+  if (desiredPresent) {
+    try {
+      await fsPromises.symlink(resources.executable(name), linkPath);
+    } catch (error: any) {
+      console.warn(`Failed to create symlink ${ linkPath }: ${ error.message }`);
+    }
+  } else if (await isManagedIntegration(linkPath)) {
+    try {
+      await fsPromises.unlink(linkPath);
+    } catch (error: any) {
+      console.error(`Error unlinking symlink ${ linkPath }: ${ error.message }`);
+    }
   }
+}
 
-  if (state) {
-    err = await new Promise((resolve) => {
-      fs.symlink(resources.executable(name), linkPath, 'file', resolve);
-    });
+/**
+ * Tests whether a path is an integration symlink that is safe to delete.
+ * @param path -- absolute path to the filesystem node that may be an integration symlink
+ */
+async function isManagedIntegration(path: string): Promise<boolean> {
+  // function is not valid if not running as appimage
+  if (!process.env['APPIMAGE']) {
+    throw new Error('isManagedIntegration is only valid when running as AppImage');
+  }
 
-    if (err) {
-      console.error(`Error creating symlink for ${ linkPath }: ${ err.message }`);
+  let linkedTo: string;
 
-      return err;
+  try {
+    linkedTo = await fsPromises.readlink(path);
+  } catch (error: any) {
+    if (error.code === 'EINVAL') {
+      return false;
     }
-  } else {
-    err = await new Promise((resolve) => {
-      fs.unlink(linkPath, resolve);
-    });
+    throw error;
+  }
 
-    if (err) {
-      console.error(`Error unlinking symlink for ${ linkPath }: ${ err.message }`);
+  const mountPath = process.env['APPDIR'];
 
-      return err;
+  if (mountPath) {
+    if (linkedTo.includes(mountPath)) {
```

Comment:
Shouldn't this be `startsWith` instead of `includes`? And maybe make sure that `mountPath` has a trailing slash?

Reply:
I agree, `startsWith` would be better - I'll make that change.

As for the trailing slash, I'm not sure what checking for it will get us. What is your rationale here?

Reply:
If `mountPath` is `/tmp` and `linkedTo` is `/tmpdir/file`, then `linkedTo.startsWith(mountPath)` will return `true` even though that is wrong.

If `mountPath` is `/tmp/` then `linkedTo.startsWith(mountPath)` properly returns `false`.

Now, I agree that this probably cannot happen here, but I think it is good practice to make code handle edge cases correctly, even when they currently can't happen.